### PR TITLE
os/drivers: refine dma update code of lcd driver

### PR DESCRIPTION
**Pull request by SRCN-LiQin**
putdma interface of lcd driver was added for fast data transmit.
refinement in this PR include:
1. support bpp < 8 device
2. use uint32 instead of uint64 to optimize mem copy
@gcjjyy

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [56fe3f587d9f44e45da3ad39fc9bd29131f897ad] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [56fe3f587d9f44e45da3ad39fc9bd29131f897ad] - Code Rule Check Result:
os/drivers/lcd/lcd_framebuffer.c:106: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:106: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/lcd_framebuffer.c:113: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:113: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/lcd_framebuffer.c:115: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:115: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/lcd_framebuffer.c:123: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:123: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/lcd_framebuffer.c:125: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:125: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/lcd_framebuffer.c:134: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:134: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/lcd_framebuffer.c:136: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:136: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/lcd_framebuffer.c:190: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:190: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/lcd_framebuffer.c:313: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:313: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/lcd_framebuffer.c:343: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:343: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/lcd_framebuffer.c:374: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:374: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/lcd_framebuffer.c:406: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:406: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/lcd_framebuffer.c:438: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:438: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/lcd_framebuffer.c:471: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:471: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/drivers/lcd/lcd_framebuffer.c:760: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/drivers/lcd/lcd_framebuffer.c:760: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Target : [738376dafcf00c170eebd74330a41857df3cf395] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [738376dafcf00c170eebd74330a41857df3cf395] - Code Rule Check OK.

File: os/drivers/lcd/lcd_framebuffer.c [hunk 1]
```diff
@@ -234,45 +241,46 @@ static int lcdfb_update(FAR struct lcdfb_dev_s *priv,
...
-
-		int offset = 0;
-		uint8_t *dmabuff = (uint8_t *)malloc(width * height * bytepp);
+		/* Send buffer with dma , need to convert buffer 
+		 * buffer for each line is byte aligned.
+		 */		
+		bytespl = (width * pinfo->bpp + 7) >> 3;
+		height = endy - starty + 1;
+		
+		uint8_t *dmabuff = (uint8_t *)malloc(height * bytespl);
```
Comment (gcjjyy) [inline os/drivers/lcd/lcd_framebuffer.c]: TizenRT's malloc and free have a lot of costs. Do you have any other method to remove malloc / free?

**Comment by seinfra** [comment]
Target : [4c9d1d44d6cfff881accd395e3f3ee939be75415] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4c9d1d44d6cfff881accd395e3f3ee939be75415] - Code Rule Check OK.

File: os/drivers/lcd/lcd_framebuffer.c (hunk 1 above)
Comment (SRCN-LiQin) [reply os/drivers/lcd/lcd_framebuffer.c]: I amended another refinement for full width update, but there isn't a suitable way to remove malloc / free here, I think a long time occupied buffer which have big size is not a good idea for the updating procedure which is called on command, although we may apply it for performance optimizing.

**Pull request merged**